devito/data/allocators.py

```diff
@@ -334,6 +337,31 @@ def put_local(self):
         return self._node == 'local'
 
 
+class PetscMemoryAllocator(MemoryAllocator):
+    """
+    """
+    @classmethod
+    def initialize(cls):
+        from devito.petsc.utils import core_metadata
+        metadata = core_metadata()
+        lib_dir = Path(metadata['lib_dirs'][-1])
+
+        try:
+            cls.lib = ctypes.CDLL(lib_dir/'libpetsc.so')
```

Comment:
Is this correct? Looks like division by a string unless I'm missing something

Reply:
lib_dir is a Path so maybe it overrides division to allow that